docs/testing/README.md

```diff
@@ -0,0 +1,124 @@
+# Testing
+
+## How to test
+
+Follow the [Guiding Principles of React Testing Library](https://github.com/testing-library/react-testing-library#guiding-principles) when writing tests.
+
+## What to test
+
+### Accessibility
+
+Automated accessibility tests are required for significant component variations and should use existing `toHaveNoAxeViolations()` and `toHaveNoDAPViolations()` Jest matchers.
+
+Use this structure for automated accessibility tests:
+
+```
+test('should have no Axe or DAP violations', async () => {
+  const main = document.createElement('main');
+  render(
+    <Component />,
+    {
+      // DAP requires a landmark '<main>' in the DOM:
+      container: document.body.appendChild(main),
+    }
+  );
+
+  await expect(document.body).toHaveNoAxeViolations();
+  await expect(document.body).toHaveNoDAPViolations('ComponentName');
+});
+
+
+test('should have no Axe or DAP violations with component variation', async () => {
+  const main = document.createElement('main');
+  render(
+    <Component variationProp={value} />,
+    {
+      // DAP requires a landmark '<main>' in the DOM:
+      container: document.body.appendChild(main),
+    }
+  );
+
+  await expect(document.body).toHaveNoAxeViolations();
+  await expect(document.body).toHaveNoDAPViolations('ComponentName with variation');
+});
+```
+
+#### Test structure
+
+- Each significant variation of a component should have its own test block.
+- Use an **asynchronous** test block, and **await** each accessibility rule matcher.
+
+#### DAP requirements
+
+- DAP requires that components be rendered inside a valid landmark element, which is why the `container` is a `main` node. This extra step may not be required if the component is already wrapped in a `main` or another significant HTML landmark element.
```

Comment:
I'm curious about this part here - have we identified many components in which this step is not required?

It seems like as we're testing components in isolation, this may never be the case - what are your thoughts?

Reply:
My suspicion is that the `Shell` is the biggest example. But either way I think it is worthwhile to keep defining a `main` wrapper... even if we had to explicitly do it for every component... because it forces the test writer to understand how/why/what DAP requires here. If we were to ever abstract that away, then we'd be basically ignoring the landmark rule & keeping it out of sight, out of mind.